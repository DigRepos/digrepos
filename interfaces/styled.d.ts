import styled, * as styledComponents from 'styled-components'

const { 
    default: styled,
    keyframes
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  IThemeInterface
>

export default styled
export { keyframes }
