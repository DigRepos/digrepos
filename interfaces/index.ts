
export type Owner = {
  name: string,
  avatarUrl: string
}

export type RepositoryData = {
  id: string,
  url: string,
  avatarUrl: string,
  fullName: string,
  description: string,
  homepage: string,
  owner: Owner,
  star: number,
  forksCount: number,
  watchersCount: number,
  topics: string[],
  language: string,
  updatedAt: string
}

export type ModalOverlayStyle = {
  colorHex: string
}

export type ModalStyle = {
  height: string,
  width: string,
  backgroundColorHex: string
}

export type SearchFilterModel = {
  keywords: string[],
  star: {
    low: string,
    high: string
  },
  language: string,
  license: string
}

export type TabKey = 'filter' | 'sort'

export type DraggableItem = {
  idx: string,
  expr: string
}
