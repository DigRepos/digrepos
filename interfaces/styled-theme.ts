import * as styledComponents from 'styled-components'

const { 
    default: styled,
    keyframes
} = styledComponents as styledComponents.ThemedStyledComponentsModule<
  any
>

export default styled
export { keyframes }