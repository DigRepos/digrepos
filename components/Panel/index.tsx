import * as React from "react"
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
import { faEye, faStar, faCodeBranch } from "@fortawesome/free-solid-svg-icons"
import { Owner } from "../../interfaces/index"
import styled from "../../interfaces/styled-theme"

type Props = {
  url: string
  avatarUrl: string
  fullName: string
  description: string
  homepage: string
  owner: Owner
  star: number
  forksCount: number
  watchersCount: number
  topics: string[]
  language: string
  updatedAt: string
}

type UserImageProps = {
  url: string
}

const PanelFrame = styled.article`
  width: auto;
  height: auto;
  padding: 8px;
  box-shadow: 5px 5px 5px 5px rgba(238, 238, 238, 1);
  display: flex;
`
const ImageArea = styled.div`
  margin: 16px 12px;
  width: 16%;
  height: 72px;
`
const OwnerImage = styled.img`
  border-radius: 4px;
  width: 72px;
  height: 72px;
  background-image: url(${(props: UserImageProps) => props.url});
  background-size:  cover;
`

const InfoArea = styled.section`
  margin: 8px;
  padding: 4px;
  width: 85%;
`
const RepositoryName = styled.div`
  margin: 4px 0px;
  word-wrap: break-word;
  font-weight: bold;
  font-size: 110%;
`
const AnchorLink = styled.a`
  text-decoration: none;
  color: #337AB7;
  :hover {
    color: #8C9EFF;
  }
`

const Description = styled.div`
  word-wrap: break-word;
`
const Homepage = styled.div`
  margin: 2px;
`
const Indexes = styled.section`
  margin: 2px;
  padding: 2px;
  display: flex;
`
const IconAndNumber = styled.div`
  margin: 4px;
`

const CounterSpan = styled.span`
  margin-left: 4px;
`
const TopicsArea = styled.section`
  margin: 2px;
  padding: 2px;
  display: flex;
  flex-wrap: wrap;
`
const TopicTag = styled.div`
  border-radius: 8px;
  margin: 2px;
  padding: 4px;
  font-size: 80%;
  background-color: #eeeeee;
`

const UpdateDate = styled.div`
  margin: 8px;
  margin-left: 60%;
  font-size: 80%;
  color: #757575;
`

const Panel: React.FC<Props> = props => {
  const makeTopics = (topics: string[]) => {
    return topics.map((v: string, i: number) => (
      <TopicTag key={i}>{v}</TopicTag>
    ))
  }

  const makeUpdateDate = (dateTime: string) => {
    return dateTime.split(" ")[0]
  }

  return (
    <PanelFrame>
      <ImageArea>
        <OwnerImage url={props.owner.avatarUrl} />
      </ImageArea>
      <InfoArea>
        <RepositoryName>
          <AnchorLink href={props.url} target={"_blank"}>
            {props.fullName}
          </AnchorLink>
        </RepositoryName>
        {props.description != "" ? (
          <Description>{props.description}</Description>
        ) : (
          "No description"
        )}
        {props.homepage != "" ? (
          <Homepage>
            <AnchorLink href={props.homepage} target={"_blank"}>
              {props.homepage}
            </AnchorLink>
          </Homepage>
        ) : (
          <></>
        )}
        <Indexes>
          <IconAndNumber>
            <FontAwesomeIcon icon={faEye} size={"sm"} />
            <CounterSpan>{props.watchersCount}</CounterSpan>
          </IconAndNumber>
          <IconAndNumber>
            <FontAwesomeIcon icon={faStar} size={"sm"} />
            <CounterSpan>{props.star}</CounterSpan>
          </IconAndNumber>
          <IconAndNumber>
            <FontAwesomeIcon icon={faCodeBranch} size={"sm"} />
            <CounterSpan>{props.forksCount}</CounterSpan>
          </IconAndNumber>
        </Indexes>
        <TopicsArea>{makeTopics(props.topics)}</TopicsArea>
        <UpdateDate>updated at {makeUpdateDate(props.updatedAt)}</UpdateDate>
      </InfoArea>
    </PanelFrame>
  )
}

export default Panel
