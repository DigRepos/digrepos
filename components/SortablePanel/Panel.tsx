import * as React from "react"
import styled from "../../interfaces/styled-theme"
import {
  useDrag,
  useDrop,
  DropTargetMonitor
} from "react-dnd-cjs"
import { XYCoord } from "dnd-core"
import ItemType from "./ItemType"
import { DraggableItem } from "../../interfaces"

const StyledPanel = styled.div`
  padding: 8px;
  margin: 12px 0px;
  text-align: center;
  border-style: solid;
  border-width: 0.8px;
  border-radius: 3px;
  color: #0ee5c7;
  border-color: #0ee5c7;
  cursor: grab;
  :active {
    cursor: grabbing;
    opacity: 0.5;
  }
`

type Props = {
  item: DraggableItem
  update: (dragIdx: number, hoverIdx: number) => void
}

const Panel: React.FC<Props> = props => {
  const ref = React.useRef<HTMLDivElement>(null)
  const [, drop] = useDrop<DraggableItem & { type: string }, any, any>({
    accept: ItemType.PANEL,
    hover(item: DraggableItem, monitor: DropTargetMonitor): void {
      if (!ref.current) {
        return
      }
      const dragIndex = item.idx
      const hoverIndex = props.item.idx

      if (dragIndex === hoverIndex) {
        return
      }

      const hoverBoundingRect = ref.current.getBoundingClientRect()

      const hoverMiddleY =
        (hoverBoundingRect.bottom - hoverBoundingRect.top) / 2

      const clientOffset = monitor.getClientOffset()

      const hoverClientY = (clientOffset as XYCoord).y - hoverBoundingRect.top
      if (dragIndex < hoverIndex && hoverClientY < hoverMiddleY) {
        return
      }

      if (dragIndex > hoverIndex && hoverClientY > hoverMiddleY) {
        return
      }
      
      props.update(dragIndex, hoverIndex)
      item.idx = hoverIndex
    }
  })
  const [, drag] = useDrag({
    item: {
      type: ItemType.PANEL,
      idx: props.item.idx,
    },
    collect: monitor => ({
      isDragging: monitor.isDragging()
    })
  })

  drag(drop(ref))

  return <StyledPanel ref={ref}>{props.item.expr}</StyledPanel>
}

export default Panel
