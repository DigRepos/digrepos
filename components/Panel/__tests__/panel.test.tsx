import * as React from 'react'
import Panel from '../index'
import renderer from 'react-test-renderer'
import { RepositoryData } from '../../../interfaces'

it('Panel component', () => {
    const props: RepositoryData = {
        id: '123456',
        url: 'aaa',
        avatarUrl: 'hoge',
        fullName: 'repository full name',
        description: 'this is test render for "Panel component"',
        homepage: 'http://www.hogefuga.com/',
        owner: {
            name: 'test user',
            avatarUrl: 'test'
        },
        star: 100,
        forksCount: 120,
        watchersCount: 25,
        topics: ['panel', 'component', 'render'],
        language: 'Java',
        updatedAt: '2019/12/11 11:24:39'
    }
    const tree = renderer.create(<Panel {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
})