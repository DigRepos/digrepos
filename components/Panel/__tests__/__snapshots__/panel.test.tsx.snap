// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel component 1`] = `
<div
  className="sc-bdVaJa dylWiv"
>
  <div
    className="sc-bwzfXH cnwBjN"
  >
    <img
      className="sc-htpNat bhgCYs"
    />
  </div>
  <div
    className="sc-bxivhb beAtmz"
  >
    <div
      className="sc-ifAKCX fCQOFr"
    >
      <a
        className="sc-EHOje hcHWUw"
        href="aaa"
        target="_blank"
      >
        repository full name
      </a>
    </div>
    <div
      className="sc-bZQynM jaDoPl"
    >
      this is test render for "Panel component"
    </div>
    <div
      className="sc-gzVnrw lfOPXV"
    >
      <a
        className="sc-EHOje hcHWUw"
        href="http://www.hogefuga.com/"
        target="_blank"
      >
        http://www.hogefuga.com/
      </a>
    </div>
    <div
      className="sc-htoDjs esImrL"
    >
      <div
        className="sc-dnqmqq fwaezI"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-eye fa-w-18 fa-sm "
          data-icon="eye"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="sc-iwsKbI bshbDz"
        >
          25
        </span>
      </div>
      <div
        className="sc-dnqmqq fwaezI"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-star fa-w-18 fa-sm "
          data-icon="star"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="sc-iwsKbI bshbDz"
        >
          100
        </span>
      </div>
      <div
        className="sc-dnqmqq fwaezI"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-code-branch fa-w-12 fa-sm "
          data-icon="code-branch"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 384 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 144c0-44.2-35.8-80-80-80s-80 35.8-80 80c0 36.4 24.3 67.1 57.5 76.8-.6 16.1-4.2 28.5-11 36.9-15.4 19.2-49.3 22.4-85.2 25.7-28.2 2.6-57.4 5.4-81.3 16.9v-144c32.5-10.2 56-40.5 56-76.3 0-44.2-35.8-80-80-80S0 35.8 0 80c0 35.8 23.5 66.1 56 76.3v199.3C23.5 365.9 0 396.2 0 432c0 44.2 35.8 80 80 80s80-35.8 80-80c0-34-21.2-63.1-51.2-74.6 3.1-5.2 7.8-9.8 14.9-13.4 16.2-8.2 40.4-10.4 66.1-12.8 42.2-3.9 90-8.4 118.2-43.4 14-17.4 21.1-39.8 21.6-67.9 31.6-10.8 54.4-40.7 54.4-75.9zM80 64c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16zm0 384c-8.8 0-16-7.2-16-16s7.2-16 16-16 16 7.2 16 16-7.2 16-16 16zm224-320c8.8 0 16 7.2 16 16s-7.2 16-16 16-16-7.2-16-16 7.2-16 16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <span
          className="sc-iwsKbI bshbDz"
        >
          120
        </span>
      </div>
    </div>
    <div
      className="sc-gZMcBi epVgsD"
    >
      <div
        className="sc-gqjmRU gjyZVQ"
      >
        panel
      </div>
      <div
        className="sc-gqjmRU gjyZVQ"
      >
        component
      </div>
      <div
        className="sc-gqjmRU gjyZVQ"
      >
        render
      </div>
    </div>
  </div>
</div>
`;
